reference-gen/go.mod

```diff
@@ -1,25 +1,27 @@
 module github.com/oauth2-proxy/tools/reference-gen
 
-go 1.22
+go 1.23.0
+
+toolchain go1.24.1
```

Comment:
Switching to 1.24 is a little too soon for my liking

Reply:
First of all, golangci-lint isn't yet compatible with your change:
https://github.com/oauth2-proxy/tools/actions/runs/13786253316/job/39002341332

Go 1.24 was just released a couple of weeks ago and if you have ever worked with larger projects that other depend upon you never want to update to a new major version immediately after it was released.

A lot of people have custom builds and forks of oauth2-proxy and we will definitely cause some unforeseen side effects if we upgrade to soon. As you can tell with golangci-lint.

Reply:
Doesn't really matter too much for the tool repo but I will keep off upgrading oauth2-proxy to v1.24 any time soon

Reply:
I'll just remove the `toolchain` line and see how it goes...

Reply:
```console
$ go mod edit -toolchain=go1.23.7
```